tools/cli/commands/changelog.js

```diff
@@ -512,18 +518,50 @@ async function gitAdd( argv ) {
 }
 
 /**
- * Gets list of currently modified files.
+ * Checks if changelog files are required.
  *
- * @returns {Array} modifiedProjects - projects that need a changelog.
+ * @returns {Array} matchedProjects - projects that need a changelog.
  */
-async function changedProjects() {
+async function checkChangelogFiles() {
+	console.log( chalk.green( 'Checking if changelog files are needed. Just a sec...' ) );
+
+	// Bail if we're pushing to a release branch, like boost/branch-1.3.0
+	let currentBranch = child_process.spawnSync( 'git', [ 'branch', '--show-current' ] );
+	currentBranch = currentBranch.stdout.toString().trim();
+	const branchReg = /.*\/branch-(\d+).(\d+)(.(\d+))?/; // match example: jetpack/branch-1.2.3
+	if ( currentBranch.match( branchReg ) ) {
+		console.log( chalk.green( 'Release branch detected. No changelog required.' ) );
+		return [];
+	}
+
 	const re = /^projects\/([^/]+\/[^/]+)\//; // regex matches project file path, ie 'project/packages/connection/..'
 	const modifiedProjects = new Set();
-	const git = simpleGit();
-	const gitStatus = await git.status();
-	for ( const file of gitStatus.files ) {
-		const match = file.path.match( re );
-		if ( match ) {
+	const changelogsAdded = [];
+	let touchedFiles = child_process.spawnSync( 'git', [
+		'-c',
+		'core.quotepath=off',
+		`diff`,
+		`--no-renames`,
+		`--name-only`,
+		`--merge-base`,
+		`origin/master`,
+	] );
+	touchedFiles = touchedFiles.stdout.toString().trim().split( '\n' );
+
+	for ( const file of touchedFiles ) {
+		const match = file.match( re );
+		if (
+			match &&
+			file.includes( `${ match[ 1 ] }/changelog/` ) &&
```

Comment:
You could drop this line (and make the checking slightly more robust) by using a different regex on line 552: `/^projects\/([^/]+\/[^/]+)\/changelog\//`.

If we really want to handle all possible cases, though, you should build `modifiedProjects` first, then loop through it, read the project's composer.json, and only then check for files starting with `path.join( 'projects', project, composerJson.extra?.['changes-dir'] ?? "changelog", "./" )` (no need for a regex there, `touchedFiles.some( file => file.startsWith( prefix ) )` would do it). But I doubt we'll ever set `.extra.changes-dir` in any project in the monorepo, so maybe that's overkill.

Reply:
> You could drop this line (and make the checking slightly more robust) by using a different regex on line 552: /^projects\/([^/]+\/[^/]+)\/changelog\//.

Went with this method for now.

Maybe add the extreme-ultra-overkill method of checking in a follow up?

Reply:
We could YAGNI the overkill method as far as I'm concerned.